src/account.js

```diff
@@ -70,11 +71,16 @@ class Account {
     ])
 
     if (this.getChannel()) {
-      // hold empty paychan details if the channel no longer exists.
-      // the channel will be cleaned up after failing validation.
-      this._paychan = await this._api.getPaymentChannel(this.getChannel())
-        .catch(() => ({}))
-      this._lastClaimedAmount = this._paychan.balance
+      try {
+        this._paychan = await this._api.getPaymentChannel(this.getChannel())
+        this._lastClaimedAmount = this._paychan.balance
+      } catch (e) {
+        debug('failed to load channel entry. error=' + e.message)
+        if (e.name === 'RippledError' && e.message === 'entryNotFound') {
+          debug('removing channel because it has been deleted')
+          this.deleteChannel()
```

Comment:
This deletes the channel id and the last claimed amount. Should the balance and the incoming claim also be deleted? When a new paychan is opened the balance should be 0. The old incoming claim is not applicable to the new channel.

Reply:
Before deleting the channel, can we check that all liabilities against the old paychan have been collected (i.e. best claim was submitted)?

Under normal operations, it should not happen that a channel can be closed without the server submitting its best claim. It would be great, however, to have a sanity check here that checks if the last claim was submitted and raise the alarm bells otherwise. Making the server delete the incoming channel without submitting the best claim has been a source of vulnerabilities in the past.

Reply:
Applied these suggestions